File: python_sdk/src/reality_capture/specifications/point_cloud_optimization.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class PCOptimizationInputs(BaseModel):
    point_clouds: list[str] = Field(description="Reality data Ids of point cloud(s) to convert", alias="pointClouds")


class PCOptimizationFormat(Enum):
    OPC = "OPC"
    THREE_D_TILES_PNTS = "3DTilesPnts"
    THREE_D_TILES_GLBC = "3DTilesGlbc"
    LAS = "LAS"
    LAZ = "LAZ"
    E57 = "E57"
    POD = "POD"


class PCOptimizationOptions(BaseModel):
    output_format: Optional[PCOptimizationFormat] = Field(None, description="Output format for the conversion.",
                                                       alias="outputFormat")
    input_crs: Optional[str] = Field(None, description="CRS for the input data", alias="inputCrs")
    output_crs: Optional[str] = Field(None, description="CRS for the output data", alias="outputCrs")


class PCOptimizationSpecificationsCreate(BaseModel):
    inputs: PCOptimizationInputs = Field(description="Inputs")
    options: Optional[PCOptimizationOptions] = Field(None, description="Options")


class PCOptimizationSpecifications(BaseModel):
    inputs: PCOptimizationInputs = Field(description="Inputs")
    output: str = Field(description="Reality Data id of the converted point cloud")
    options: Optional[PCOptimizationOptions] = Field(None, description="Options")

File: typescript/packages/reality-capture/src/specifications/point_cloud_optimization.ts
import { z } from "zod";

export const PCOptimizationInputsSchema = z.object({
  pointClouds: z
    .array(z.string())
    .describe("Reality data Ids of point cloud(s) to convert"),
});
export type PCOptimizationInputs = z.infer<typeof PCOptimizationInputsSchema>;

export enum PCOptimizationFormat {
  OPC = "OPC",
  THREE_D_TILES_PNTS = "3DTilesPnts",
  THREE_D_TILES_GLBC = "3DTilesGlbc",
  LAS = "LAS",
  LAZ = "LAZ",
  E57 = "E57",
  POD = "POD",
}

export const PCOptimizationOptionsSchema = z.object({
  outputFormat: z
    .nativeEnum(PCOptimizationFormat)
    .describe("Output format for the conversion.")
    .optional(),
  inputCrs: z.string().describe("CRS for the input data").optional(),
  outputCrs: z.string().describe("CRS for the output data").optional(),
});
export type PCOptimizationOptions = z.infer<typeof PCOptimizationOptionsSchema>;

export const PCOptimizationSpecificationsCreateSchema = z.object({
  inputs: PCOptimizationInputsSchema.describe("Inputs"),
  options: PCOptimizationOptionsSchema.describe("Options").optional(),
});
export type PCOptimizationSpecificationsCreate = z.infer<
  typeof PCOptimizationSpecificationsCreateSchema
>;

export const PCOptimizationSpecificationsSchema = z.object({
  inputs: PCOptimizationInputsSchema.describe("Inputs"),
  output: z.string().describe("Reality Data id of the converted point cloud"),
  options: PCOptimizationOptionsSchema.describe("Options").optional(),
});
export type PCOptimizationSpecifications = z.infer<
  typeof PCOptimizationSpecificationsSchema
>;
